lib/ansible/modules/network/eos/eos_static_route.py

```diff
@@ -0,0 +1,212 @@
+        admin_distance = w['admin_distance']
+        state = w['state']
+        del w['state']
+
+        if state == 'absent' and w in have:
+            commands.append('no ip route %s %s' % (address, next_hop))
+        elif state == 'present' and w not in have:
+            commands.append('ip route %s %s %d' % (address, next_hop, admin_distance))
+
+    return commands
+
+
+def map_params_to_obj(module, required_together=None):
+    obj = []
+
+    aggregate = module.params.get('aggregate')
+    if aggregate:
+        for item in aggregate:
+            for key in item:
+                if item.get(key) is None:
+                    item[key] = module.params[key]
+
+            module._check_required_together(required_together, item)
+            d = item.copy()
+
+            obj.append(d)
+    else:
+        obj.append({
+            'address': module.params['address'].strip(),
+            'next_hop': module.params['next_hop'].strip(),
+            'admin_distance': module.params['admin_distance'],
+            'state': module.params['state']
+        })
+
+    return obj
+
+
+def map_config_to_obj(module):
+    objs = []
+
+    try:
+        out = run_commands(module, ['show ip route | json'])[0]
+    except IndexError:
+        out = {}
+    if out:
+        try:
+            vrfs = out['vrfs']['default']['routes']
+        except (AttributeError, KeyError, TypeError):
+            vrfs = {}
+    if vrfs:
+        for address in vrfs:
+            obj = {}
+            obj['address'] = address
+            obj['admin_distance'] = vrfs[address].get('preference')
+            obj['next_hop'] = vrfs[address].get('vias')[0].get('nexthopAddr')
+            objs.append(obj)
+
+    return objs
+
+
+def main():
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        address=dict(type='str', aliases=['prefix']),
+        next_hop=dict(type='str'),
+        admin_distance=dict(default=1, type='int'),
+        state=dict(default='present', choices=['present', 'absent'])
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['address'] = dict(required=True)
+
+    # remove default in aggregate spec, to handle common arguments
+    remove_default_spec(aggregate_spec)
+
+    argument_spec = dict(
+        aggregate=dict(type='list', elements='dict', options=aggregate_spec),
+    )
+
+    argument_spec.update(element_spec)
+    argument_spec.update(eos_argument_spec)
+
+    required_one_of = [['aggregate', 'address']]
+    required_together = [['address', 'next_hop']]
+    mutually_exclusive = [['aggregate', 'address']]
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_one_of=required_one_of,
+                           required_together=required_together,
+                           mutually_exclusive=mutually_exclusive,
+                           supports_check_mode=True)
+
+    warnings = list()
+    check_args(module, warnings)
```

Comment:
Verifying here that IP address and prefix are in correct format would be good.
Look for APIs:
`socket.inet_pton(socket.AF_INET, address)` and `socket.inet_pton(socket.AF_INET6, address)`